### PR TITLE
Updated computation UI for latest version of Swift.

### PR DESCRIPTION
**Pull request by Brow71189**

### CODE REVIEW

File: nionswift_plugin/nion_experimental_4dtools/ComputationUI.py [hunk 1]
```diff
@@ -59,15 +72,15 @@ def __display_item_changed(self, display_item):
                 for result in computation.results:
                     if result.specifier.get('uuid') == str(data_item.uuid):
                         computations_involved.append(computation)
-                if computation.source:
-                    if str(computation.source.uuid) == str(data_item.uuid):
-                        computations_involved.append(computation)
+                src = computation.get_input('src')
+                if src and str(src.uuid) == str(data_item.uuid):
+                    computations_involved.append(computation)
```
Comment (cmeyer) [inline nionswift_plugin/nion_experimental_4dtools/ComputationUI.py]: The 'src' and `.source` are not the same thing. One refers to another object in Swift that "owns" the computation. The other refers to the variable named "src" in the computation inputs. Not all computations will have "src" as an input. Is this change intentional?

File: nionswift_plugin/nion_experimental_4dtools/ComputationUI.py [hunk 2]
```diff
@@ -59,15 +72,15 @@ def __display_item_changed(self, display_item):
...
 
             self.__update_computation_ui(computations_involved)
             for _ in computations_involved:
                 self.__computation_updated_event_listeners.append(
                         self.document_controller._document_controller.document_model.computation_updated_event.listen(
-                                lambda data_item, new_computation: self.__update_computation_ui(computations_involved)))
+                                lambda data_item: self.__update_computation_ui(computations_involved)))
```
Comment (cmeyer) [inline nionswift_plugin/nion_experimental_4dtools/ComputationUI.py]: If we make it `lamdba data_item, *args, **kwargs` then it will work with both 0.14 and 0.15 I think. I haven't tried it, though.

File: nionswift_plugin/nion_experimental_4dtools/ComputationUI.py [hunk 3]
```diff
@@ -32,6 +35,16 @@ def __update_computation_ui(self, computations):
...
+            if create_panel_widget is None:
+                compute_class = Symbolic._computation_types.get(computation.processing_id)
+                if compute_class:
+                    api_computation = self.api._new_api_object(computation)
+                    api_computation.api = self.api
+                    try:
+                        compute_class(api_computation)
+                    except IndexError:
+                        pass
+                    create_panel_widget = getattr(computation, 'create_panel_widget', None)
```
Comment (cmeyer) [inline nionswift_plugin/nion_experimental_4dtools/ComputationUI.py]: What's the purpose of this section? (the reason I'm asking is if we don't need it, then the only change should be the  lambda function below -- and with appropriate arguments to that lambda, we can make this code compatible with both 0.14 and 0.15 for the time being...?

Comment (Brow71189) [reply nionswift_plugin/nion_experimental_4dtools/ComputationUI.py]: I forgot about this part in my pull request. The purpose is to show the computation UI after a re-start of Swift.
The `create_panel_widget` attribute gets set in the `__init__` method of the computation class, so we need to instantiate it before we can access it. After restarting Swift, computations will only be instantiated when something in them changes, but for some of them you cannot change anything without the UI, so they would never get instantiated.

File: nionswift_plugin/nion_experimental_4dtools/ComputationUI.py (hunk 1 above)
Comment (Brow71189) [reply nionswift_plugin/nion_experimental_4dtools/ComputationUI.py]: I wasn't aware of the exact difference between the two and I don't think it is beneficial to use 'src' over '.source' so I changed it back.

File: nionswift_plugin/nion_experimental_4dtools/ComputationUI.py (hunk 2 above)
Comment (Brow71189) [reply nionswift_plugin/nion_experimental_4dtools/ComputationUI.py]: Agreed, I will change and test it.
